### DISCUSSION TITLE
Flat relational fields on api

### DISCUSSION THREAD

**Discussion by alishi973** [directus/directus#17589] [Archive]
Hi, maybe this question asked for a lot time, but this is not clear to me what the solution is and how to solve this problem:

We have a simple product model like this:

```typescript
type Product {
   id: string
   name: string
   price: string
   images: string[]
}
```
Where the `images` array contains the `id` of the image from the `product_image` model:

```typescript
type ProductImage {
   id: string
   product: string
   image: string
}
```

so when we want to retrieve the specific product, the result is something like this:

```js
// with this query: xyz.com/items/products/:id?fields[]=*,images.image

{
     id: "1234-5678-90123",
     price: 100,
     name: "blah blah",
     /* ... */
     images: [
          {
               image: "221fe7e3-6416-4843-975a-3eeb27c628e2"
          },
          {
               image: "c43dc7e0-95c5-4aa5-9176-491ddc2e0caf"
          },
          {
               image: "1b5ea6fe-68e2-492b-84db-a4074616f14f"
          },
          {
               image: "ef92e755-6872-4da2-9b6b-5134e297c970"
          },
          {
               image: "457898c5-1a0f-4287-bd34-428c6e9c3076"
          }
     ]
}
```

My question is, Is there any option to return only the value of `images.image` instead of returning the object?

**Comment by br41nslug** [comment]
You'd have to either format the returned object on the frontend or create a custom endpoint to format it on the backend :)

**Reply by alishi973** [reply]
So does it means that we do not have any plan to provide a solution for such a case? because it seems for all types of relational fields we have problems like this.

**Reply by br41nslug** [reply]
Directus is mirroring the data structure from your database as described in the docs https://docs.directus.io/getting-started/introduction.html#how-it-works
There are a couple solutions available for this are as mentioned in my previous post:
- reformat the data on the frontend after receiving it from the API
- create custom extension to reformat the data directly in Directus (possible using either a custom endpoint or custom hook)
- restructure your data-model to reflect the desired structure

If you have other ideas/suggestion you're always welcome to create a feature request :D
